tests/conftest.py

```diff
@@ -116,7 +162,10 @@ def teardown():
                     # just manually retry the flushdb
                     client.flushdb()
             client.close()
-            client.connection_pool.disconnect()
+            if REDIS_INFO["cluster_enabled"]:
+                client.disconnect_connection_pools()
```

Comment:
Same - assume the non-clustered case as the default. But this disconnect is *GREAT*